contracts/governance/Vesting/VestingLogic.sol

```diff
@@ -222,4 +239,21 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
         selectors[0] = this.stakeTokensWithApproval.selector;
         return selectors;
     }
+
+    /**
+     * @notice Max iteration for vesting withdrawal to prevent out of gas issue.
+     *
+     * @return max iteration value.
+     */
+    function getMaxVestingWithdrawIterations() public view returns (uint256) {
+        return maxVestingWithdrawIterations;
```

Comment:
pls consider unloading the code by just querying the `maxVestingWithdrawIterations` from the Staking contract as this parameter is going to be the same across all withdrawals, so the Staking is the ultimate source of thruth

Reply:
Solved in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/f938b80ff498b0936d322885fd6e5d247407339b)